src/yunohost/service.py

```diff
@@ -415,41 +415,33 @@ def service_log(name, number=50):
         raise YunohostError('service_unknown', service=name)
 
     log_list = services[name].get('log', [])
-    log_type_list = services[name].get('log_type', [])
 
-    if not isinstance(log_list, list):
-        log_list = [log_list]
-    if len(log_type_list) < len(log_list):
-        log_type_list.extend(["file"] * (len(log_list)-len(log_type_list)))
+    # Legacy stuff related to --log_type where we'll typically have the service
+    # name in the log list but it's not an actual logfile. Nowadays journalctl
+    # is automatically fetch as well as regular log files.
+    log_list.remove(name)
 
     result = {}
 
     # First we always add the logs from journalctl / systemd
     result["journalctl"] = _get_journalctl_logs(name, number).splitlines()
 
-    for index, log_path in enumerate(log_list):
-        log_type = log_type_list[index]
+    for log_path in log_list:
+        # log is a file, read it
+        if not os.path.isdir(log_path):
```

Comment:
```suggestion
        if os.path.isfile(log_path):
```

Reply:
There's also a subtlety here which should clearly be improved as well .. but basically you want to cover the case where a service has `--log /some/dir` and the dir doesn't exists yet ...

With the current code it does work, but if check `isfile`, it will say "nope", will then proceed to the rest of the code an try to `os.listdir("/some/dir")` which will throw an expection if folder doesn't exists :s

Reply:
I don't know, maybe this code is clearer then ?

```python
if os.path.isfile(log_path):
    result[log_path] = _tail(log_path, number)
else if not os.path.isdir(log_path):
    continue
```

Reply:
Alrighty :stuck_out_tongue_winking_eye: Commited in https://github.com/YunoHost/yunohost/commit/f25e07fd829cd179393762d5c23a6f9f2670ed1c